crates/ruff_server/src/server/api/notifications/did_change_workspace.rs

```diff
@@ -21,20 +19,11 @@ impl super::SyncNotificationHandler for DidChangeWorkspace {
         params: types::DidChangeWorkspaceFoldersParams,
     ) -> Result<()> {
         for types::WorkspaceFolder { ref uri, .. } in params.event.added {
-            let workspace_path = uri
-                .to_file_path()
-                .map_err(|()| anyhow!("expected document URI {uri} to be a valid file path"))
-                .with_failure_code(ErrorCode::InvalidParams)?;
-
-            session.open_workspace_folder(workspace_path);
+            session.open_workspace_folder(uri.clone());
```

Comment:
Nit: You can remove the `clone` here by removing the `ref` from `uri`
```suggestion
        for types::WorkspaceFolder { uri, .. } in params.event.added {
            session.open_workspace_folder(uri);
```